Magnet to Main?

Gustav-Friede opened the pull request:
magnet_node.py, NODE that switches the magnet on and off during execution, based on the boolean value of the subscribed topic. (the names for subscribed and published topics may need to be adjusted)

magnet_node_pwm.py (in resource), NODE that controls the magnet with a PWM signal (to adjust the intensity of the magnetic field). (does not work)

magnet.py (in resource), switches the magnet on and off during execution based on the command console input.

jonathango-berkeley committed:
Revert "update main with subscribers & publishers"

jonathango-berkeley committed:
Revert "update main with subscribers & publishers"

jonathango-berkeley committed:
add update from 206a

jonathango-berkeley committed:
add update from 206a

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
initial

Gustav-Friede committed:
working on magnet node

Gustav-Friede committed:
working on magnet node

Gustav-Friede committed:
working on magnet node

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
nodes draft

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

Gustav-Friede committed:
freq fix

jonathango-berkeley committed:
change setup.py

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

Gustav-Friede committed:
clean up

jonathango-berkeley commented on src/sensor_interface/sensor_interface/magnet_node.py:
```diff
@@ -0,0 +1,62 @@
+        GPIO.setup(PIN, GPIO.OUT, initial=GPIO.LOW)    #setting PIN as output
+        
+        #subscriber
+        self.subscription = self.create_subscription(Bool, '/magnet_control', self.magnet_callback, 10)    #msg type? topic name?
```
once @fernandopluma is done with his branch, let's change this to the finalized topic

fernandopluma committed:
Aruco detection (#8)

* Aruco Marker detection with pose estimation

* added posestamped and /is_human topic

* added child_frame_id = aruco_marker_74

* updated files and added export DISPLAY

* clean up

* change info to debug

---------

Co-authored-by: Jonathan Goenadibrata <jonathango@berkeley.edu>

jonathango-berkeley committed:
Add Camera to Robot Launch (#9)

Create a launch file to add the camera portion of the code into the main robot_launch.py file.
* add sensor_launch.py
* added sensor_launch to robot_launch

Gustav-Friede committed:
Merge branch 'magnet' of https://github.com/jonathango-berkeley/rosmaster into magnet

Gustav-Friede committed:
Merge branch 'magnet' of https://github.com/jonathango-berkeley/rosmaster into magnet

Gustav-Friede committed:
Merge branch 'magnet' of https://github.com/jonathango-berkeley/rosmaster into magnet

Gustav-Friede committed:
Merge branch 'magnet' of https://github.com/jonathango-berkeley/rosmaster into magnet

Gustav-Friede committed:
Merge branch 'magnet' of http://github.com/jonathango-berkeley/rosmaster into magnet
